Error while checking build/O2/fullCI for 149d36257f487c096c69852a98e4766d26325c6f at 2024-04-29 20:20:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
/sw/SOURCES/O2/13096-slc8_x86-64/0/Framework/Foundation/include/Framework/Signpost.h:118:38: error: no matching function for call to 'sprintf(std::array<char, 58>, long int&)'
/sw/slc8_x86-64/fmt/10.1.1-19/include/fmt/printf.h:603:11: error: no type named 'type' in 'struct fmt::v10::detail::char_t_impl<std::array<char, 58>, void>'
/sw/SOURCES/O2/13096-slc8_x86-64/0/Framework/Foundation/include/Framework/Signpost.h:118:38: error: no matching function for call to 'sprintf(std::array<char, 58>, int64_t&)'
/sw/slc8_x86-64/fmt/10.1.1-19/include/fmt/printf.h:603:11: error: no type named 'type' in 'struct fmt::v10::detail::char_t_impl<std::array<char, 58>, void>'
/sw/SOURCES/O2/13096-slc8_x86-64/0/Framework/Foundation/include/Framework/Signpost.h:118:38: error: no matching function for call to 'sprintf(std::array<char, 40>)'
/sw/slc8_x86-64/fmt/10.1.1-19/include/fmt/printf.h:603:11: error: no type named 'type' in 'struct fmt::v10::detail::char_t_impl<std::array<char, 40>, void>'
/sw/SOURCES/O2/13096-slc8_x86-64/0/Framework/Foundation/include/Framework/Signpost.h:118:38: error: no matching function for call to 'sprintf(std::array<char, 61>, int)'
/sw/slc8_x86-64/fmt/10.1.1-19/include/fmt/printf.h:603:11: error: no type named 'type' in 'struct fmt::v10::detail::char_t_impl<std::array<char, 61>, void>'
/sw/SOURCES/O2/13096-slc8_x86-64/0/Framework/Foundation/include/Framework/Signpost.h:118:38: error: no matching function for call to 'sprintf(std::array<char, 77>, int)'
/sw/slc8_x86-64/fmt/10.1.1-19/include/fmt/printf.h:603:11: error: no type named 'type' in 'struct fmt::v10::detail::char_t_impl<std::array<char, 77>, void>'
/sw/SOURCES/O2/13096-slc8_x86-64/0/Framework/Foundation/include/Framework/Signpost.h:118:38: error: no matching function for call to 'sprintf(std::array<char, 85>)'
/sw/slc8_x86-64/fmt/10.1.1-19/include/fmt/printf.h:603:11: error: no type named 'type' in 'struct fmt::v10::detail::char_t_impl<std::array<char, 85>, void>'
/sw/SOURCES/O2/13096-slc8_x86-64/0/Framework/Foundation/include/Framework/Signpost.h:118:38: error: no matching function for call to 'sprintf(std::array<char, 74>)'
/sw/slc8_x86-64/fmt/10.1.1-19/include/fmt/printf.h:603:11: error: no type named 'type' in 'struct fmt::v10::detail::char_t_impl<std::array<char, 74>, void>'
/sw/SOURCES/O2/13096-slc8_x86-64/0/Framework/Foundation/include/Framework/Signpost.h:118:38: error: no matching function for call to 'sprintf(std::array<char, 91>)'
/sw/slc8_x86-64/fmt/10.1.1-19/include/fmt/printf.h:603:11: error: no type named 'type' in 'struct fmt::v10::detail::char_t_impl<std::array<char, 91>, void>'
/sw/SOURCES/O2/13096-slc8_x86-64/0/Framework/Foundation/include/Framework/Signpost.h:118:38: error: no matching function for call to 'sprintf(std::array<char, 75>)'
/sw/slc8_x86-64/fmt/10.1.1-19/include/fmt/printf.h:603:11: error: no type named 'type' in 'struct fmt::v10::detail::char_t_impl<std::array<char, 75>, void>'
/sw/SOURCES/O2/13096-slc8_x86-64/0/Framework/Foundation/include/Framework/Signpost.h:118:38: error: no matching function for call to 'sprintf(std::array<char, 28>)'
/sw/slc8_x86-64/fmt/10.1.1-19/include/fmt/printf.h:603:11: error: no type named 'type' in 'struct fmt::v10::detail::char_t_impl<std::array<char, 28>, void>'
/sw/SOURCES/O2/13096-slc8_x86-64/0/Framework/Foundation/include/Framework/Signpost.h:118:38: error: no matching function for call to 'sprintf(std::array<char, 32>)'
/sw/slc8_x86-64/fmt/10.1.1-19/include/fmt/printf.h:603:11: error: no type named 'type' in 'struct fmt::v10::detail::char_t_impl<std::array<char, 32>, void>'
/sw/SOURCES/O2/13096-slc8_x86-64/0/Framework/Foundation/include/Framework/Signpost.h:118:38: error: no matching function for call to 'sprintf(std::array<char, 51>)'
/sw/slc8_x86-64/fmt/10.1.1-19/include/fmt/printf.h:603:11: error: no type named 'type' in 'struct fmt::v10::detail::char_t_impl<std::array<char, 51>, void>'
/sw/SOURCES/O2/13096-slc8_x86-64/0/Framework/Foundation/include/Framework/Signpost.h:118:38: error: no matching function for call to 'sprintf(std::array<char, 38>)'
/sw/slc8_x86-64/fmt/10.1.1-19/include/fmt/printf.h:603:11: error: no type named 'type' in 'struct fmt::v10::detail::char_t_impl<std::array<char, 38>, void>'
/sw/SOURCES/O2/13096-slc8_x86-64/0/Framework/Foundation/include/Framework/Signpost.h:118:38: error: no matching function for call to 'sprintf(std::array<char, 60>, std::vector<o2::framework::DataProcessingDevice::handleData(o2::framework::ServiceRegistryRef, o2::framework::InputChannelInfo&)::InputInfo>::size_type, fair::mq::Parts::size_type)'
/sw/slc8_x86-64/fmt/10.1.1-19/include/fmt/printf.h:603:11: error: no type named 'type' in 'struct fmt::v10::detail::char_t_impl<std::array<char, 60>, void>'
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/13096/149d36257f487c096c69852a98e4766d26325c6f/build_O2_fullCI/pretty.html).

--- a/Framework/Foundation/include/Framework/Signpost.h
+++ b/Framework/Foundation/include/Framework/Signpost.h
@@ -26,9 +26,8 @@ struct o2_log_handle_t {
 
 // Helper function which replaces engineering types with a printf
 // compatible format string.
-// FIXME: make this consteval when available in C++20
 template <auto N>
-constexpr auto remove_engineering_type(char const (&src)[N])
+consteval auto remove_engineering_type(char const (&src)[N])
 {
   std::array<char, N> res = {};
   // do whatever string manipulation you want in res.
@@ -501,7 +500,7 @@ void o2_debug_log_set_stacktrace(_o2_log_t* log, int stacktrace)
   } else if (O2_BUILTIN_UNLIKELY(private_o2_log_##log->stacktrace)) {                                               \
     _o2_signpost_event_emit(private_o2_log_##log, id, name, remove_engineering_type(format).data(), ##__VA_ARGS__); \
   } else {                                                                                                          \
-    O2_LOG_MACRO_RAW(info, format, ##__VA_ARGS__);                                                                  \
+    O2_LOG_MACRO_RAW(info, remove_engineering_type(format).data(), ##__VA_ARGS__);                                  \
   }                                                                                                                 \
 })
 
@@ -512,7 +511,7 @@ void o2_debug_log_set_stacktrace(_o2_log_t* log, int stacktrace)
   } else if (O2_BUILTIN_UNLIKELY(private_o2_log_##log->stacktrace)) {                                               \
     _o2_signpost_event_emit(private_o2_log_##log, id, name, remove_engineering_type(format).data(), ##__VA_ARGS__); \
   }                                                                                                                 \
-  O2_LOG_MACRO_RAW(error, format, ##__VA_ARGS__);                                                                   \
+  O2_LOG_MACRO_RAW(error, remove_engineering_type(format).data(), ##__VA_ARGS__);                                   \
 })
 
 // Similar to the above, however it will also print a normal warning message regardless of the signpost being enabled or not.
@@ -522,7 +521,7 @@ void o2_debug_log_set_stacktrace(_o2_log_t* log, int stacktrace)
   } else if (O2_BUILTIN_UNLIKELY(private_o2_log_##log->stacktrace)) {                                               \
     _o2_signpost_event_emit(private_o2_log_##log, id, name, remove_engineering_type(format).data(), ##__VA_ARGS__); \
   }                                                                                                                 \
-  O2_LOG_MACRO_RAW(warn, format, ##__VA_ARGS__);                                                                    \
+  O2_LOG_MACRO_RAW(warn, remove_engineering_type(format).data(), ##__VA_ARGS__);                                    \
 })
 
 #define O2_SIGNPOST_START(log, id, name, format, ...)                                                                   \

